src/PasswordHasher.php

```diff
@@ -93,9 +101,31 @@ public function validate(
         string $hash
     ): bool {
         if ($password === '') {
-            throw new \InvalidArgumentException('Password must be a string and cannot be empty.');
+            throw new InvalidArgumentException('Password must be a string and cannot be empty.');
         }
 
         return password_verify($password, $hash);
     }
+
+    /**
+     * Verifies a hash needs rehash.
+     *
+     * @param string $hash The hash to verify the password against.
+     *
+     * @throws InvalidArgumentException on empty hash parameters
+     *
+     * @return bool whether the hash is need to rehash or not.
+     *
+     * @see password_needs_rehash()
+     */
+    public function needsRehash(
+        #[SensitiveParameter]
+        string $hash
+    ): bool {
+        if ($hash === '') {
```

Comment:
Why need check hash on empty? Seems, `password_needs_rehash` works with empty string correctly.

Reply:
> throw the password into the hash method in a positive result

This is expected behavior, isn't it?

Reply:
> > throw the password into the hash method in a positive result
> 
> This is expected behavior, isn't it?

A non-empty password can have an empty hash? )

Reply:
I think this check is a nice safeguard.